utils/managed_bytes.hh

```diff
@@ -451,8 +451,8 @@ struct appending_hash<managed_bytes_view> {
     template<Hasher Hasher>
     void operator()(Hasher& h, managed_bytes_view v) const {
         feed_hash(h, v.size_bytes());
-        for (; !v.empty(); v.remove_current()) {
-            h.update(reinterpret_cast<const char*>(v.current_fragment().data()), v.current_fragment().size());
+        for (bytes_view frag : fragment_range(v)) {
+            h.update(reinterpret_cast<const char*>(frag.data()), frag.size());
```

Comment:
`appending_hash` must be backwards-compatible with older Scylla versions.

Is update() linear with respect to concatenation? `update(A+B) = update(A) ; update(B)`

Reply:
This comment is on the wrong patch though. Did we hash linearized form before the series or was the problem preexisting?

Reply:
`managed_bytes_view` did not exist before the series, so it didn't have an `appending_hash` at all. And after the series it's used only to implement `std::hash` for `managed_bytes` and `managed_bytes_view`.

Also, what do you mean by `appending_hash` compatibility? It only feeds the bytes to the actual hashing algorithm.

Reply:
This series changes the hashing algorithm used in `std::hash`, but `std::hash` doesn't have to stay compatible between scylla versions, right?

Reply:
> Is update() linear with respect to concatenation? `update(A+B) = update(A) ; update(B)`

That's a precondition for the `Hasher` passed to `appending_hash`. `bytes_view_hasher`, added in this series and used to implement `std::hash` for `bytes_view`, `managed_bytes_view` and `managed_bytes`, fulfils that precondition.

Reply:
One more clarification: before the series, the keys were linearized and their linearized content was passed to `appending_hash`. After the series, they are not linearized and are passed to `appending_hash` as `managed_bytes_view`.

`appending_hash` for `bytes_view` feeds the `Hasher` all data at once, `appending_hash` for `managed_bytes_view` feeds the `Hasher` fragment by fragment. It's the responsibity of the `Hasher` to give the same result in both cases.

I didn't change the `Hasher`s that need to stay compatible, so I didn't make any incompatible changes.

I also added a new hasher, `bytes_view_hasher`, which is used to implement `std::hash` for `managed_bytes_view`.

@tgrabiec Does this answer your question?